usage/rest-client/src/main/java/org/apache/brooklyn/rest/client/BrooklynApi.java

```diff
@@ -114,11 +150,23 @@ public BrooklynApi(String endpoint, @Nullable Credentials credentials) {
         }
     }
 
+    /**
+     * @deprecated since 0.9.0. Use {@link BrooklynApiFactory#getBrooklynApi(java.net.URL, String, String)} instead
+     * @see org.apache.brooklyn.rest.client.DefaultBrooklynApiFactory
+     * @see org.apache.brooklyn.rest.client.PoolingBrooklynApiFactory
+     */
+    @Deprecated
```

Comment:
Then probably it's best to reinstate the previous public constructor and remove the new one. The subclassing is certainly too complicated, especially if we don't follow the factory route.